### PR TITLE
Links refactor

### PR DESCRIPTION
**Pull request by nnhw**

### CODE REVIEW

File: src/module/meta.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
-    "name": "Jerom-The-Mavlink",
-    "version": "0.0.1",
-    "description": "Mavlink communication module",
-    "dependencies" : ["Vegur"]
+  "name": "Jerom-The-Mavlink",
+  "version": "0.0.1",
+  "description": "Mavlink communication protocol module",
```
Comment (MishkaRogachev) [inline src/module/meta.json]: Jerom-The-Mavlink is not only protocol module, it contatins lots of non-protocol stuff

File: src/module/meta.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+  "dependencies": [
+    "Vegur"
```
Comment (MishkaRogachev) [inline src/module/meta.json]: This can be deleted

File: src/module/module_jerom_the_mavlink.cpp [hunk 3]
```diff
@@ -59,20 +59,22 @@ void ModuleJeromTheMavlink::init()
...
     missionService->registerMissionType(&domain::mission::mavlinkMissionType);
 
     domain::MavlinkHandlerFactory factory(pTree, missionService, vehiclesService, commandsService);
 
-    auto configuration = new data_source::LinkConfiguration(::linksFileName);
-    m_transceiver =
-        new domain::MavlinkTranscieverThreaded(new domain::MavlinkTransceiver(configuration,
-                                                                              &factory, nullptr),
-                                               this);
+    m_protocol = new data_source::MavlinkProtocolThreaded(new data_source::MavlinkProtocol(&factory,
```
Comment (MishkaRogachev) [inline src/module/module_jerom_the_mavlink.cpp:68]: If creation in init(), destruction must be in done(), or use shared/scoped pointer

File: src/protocols/mavlink_protocol.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    for (domain::IMavlinkHandler* handler : qAsConst(m_handlers))
+    {
+        handler->setParent(this);
+        connect(handler, &domain::IMavlinkHandler::sendMessage, this, &MavlinkProtocol::sendMessage);
+    }
+}
+
+MavlinkProtocol::~MavlinkProtocol()
+{
+    for (domain::IMavlinkHandler* handler : qAsConst(m_handlers))
```
Comment (MishkaRogachev) [inline src/protocols/mavlink_protocol.cpp]: qDeleteAll()

File: src/protocols/mavlink_protocol.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#ifndef MAVLINK_PROTOCOL_H
+#define MAVLINK_PROTOCOL_H
+
+#include "i_communication_protocol.h"
+
+#include <QMap>
+#include <QVector>
+
+#include <memory>
```
Comment (MishkaRogachev) [inline src/protocols/mavlink_protocol.h]: unneeded

File: src/protocols/mavlink_protocol.h [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+public slots:
+    void receiveData(const QByteArray& data) override;
+
+private slots:
+    void sendMessage(const mavlink_message_t& message);
+
+private:
+    void parseMessage(const QByteArray& data);
+    domain::MavlinkHandlerContext m_context;
```
Comment (MishkaRogachev) [inline src/protocols/mavlink_protocol.h:28]: new line

File: src/protocols/mavlink_protocol_threaded.cpp [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        qCritical() << "Thread" << m_thread->objectName() << "is blocked!";
+        qCritical() << "Forcing to terminate...";
+        m_thread->terminate();
+    }
+}
+
+void MavlinkProtocolThreaded::receiveData(const QByteArray& data)
+{
+    emit resendData(data);
```
Comment (MishkaRogachev) [inline src/protocols/mavlink_protocol_threaded.cpp]: dataRecived?

**Review by MishkaRogachev — CHANGES_REQUESTED** [review]
Some minor issues

File: src/module/module_jerom_the_mavlink.cpp [hunk 8]
```diff
@@ -93,5 +95,5 @@ void ModuleJeromTheMavlink::done()
     Q_ASSERT(missionService);
     missionService->unregisterMissionType(&domain::mission::mavlinkMissionType);
 
-    m_transceiver->stop();
+    delete m_protocol;
```
Comment (MishkaRogachev) [inline src/module/module_jerom_the_mavlink.cpp]: MavlinkProtocolThreaded is QObject, we can just set parent for  m_protocol

**Review by MishkaRogachev — CHANGES_REQUESTED** [review]

File: src/module/module_jerom_the_mavlink.cpp (hunk 8 above)
Comment (nnhw) [reply src/module/module_jerom_the_mavlink.cpp]: ok, MavlinkProtocolThreaded is already a QObject child for this module-object

**Review by MishkaRogachev — APPROVED** [review]

**Pull request merged**